core/modules/node/node.module

```diff
@@ -1901,6 +1909,33 @@ function node_block_view($delta = '', $settings = array()) {
         }
       }
       break;
+
+    case 'empty':
+      // Check if the promoted view exists and still has the block display.
+      $view = views_get_view('promoted');
+      if ($view && $view->access('block')) {
+        $nids = views_get_view_result('promoted', 'block');
```

Comment:
I think I mentioned this previously, but doing this double-loads the view. This is not good from a performance standpoint to be executing the view twice on the same page. This logic should be altered into the "promoted" view rather than being a separate block.